### DISCUSSION TITLE
关于在 “Xray(M+F+B+D+G+A)+Caddy(N)” 配置中添加 SplitHTTP 代理的尝试中出现的问题。

### DISCUSSION THREAD

**Discussion by rannj** [lxhao61/integrated-examples#219] [Q&A] [answered]
lxhao61大你好，首先感谢你的配置示例！通过你的配置我熟悉了Caddy和Xray的很多配置，实际运行起来体验也很好。

我尝试在你的“Xray(M+F+B+D+G+A)+Caddy(N)”配置基础上，参考“Xray(VLESS+SplitHTTP)+Nginx\Caddy”来增加SplitHTTP代理（http2和http3），我的配置和运行中的错误如下：

1.使用配置1时，由Caddy提供TLS，H2和H3均能运行并连通，但是错误日志会一直输出。其错误日志主要是：

客户端启用"alpn": ["h3"]：
http.handlers.reverse_proxy	reading from backend   "error": "stream 20 canceled by remote with error code 268"
http.handlers.reverse_proxy	aborting with incomplete response   "error": "reading: stream 20 canceled by remote with error code 268"

客户端不启用"alpn": ["h3"]：
http.handlers.reverse_proxy	aborting with incomplete response   "error": "reading: context canceled"

详细日志见附件

2.使用配置2时，由Xray提供TLS，此时若客户端不启用"alpn": ["h3"]，则正常运行且无错误，若启用，则无法连通服务器，caddy error为连接被拒绝

3.另外，我的naive配置按照示例运行，但是无法通过http3连接，caddy access中仅有http2记录

[caddy配置1(参考本项目的SplitHTTP配置).json](https://github.com/user-attachments/files/17356840/caddy.1.SplitHTTP.json)

[caddy配置2(为xray代理请求).json](https://github.com/user-attachments/files/17356841/caddy.2.xray.json)

[配置1的错误日志.log](https://github.com/user-attachments/files/17356842/1.log)

**Comment by lxhao61** [comment, marked as answer]
你配置无逻辑问题，只是不太符合优化及安全配置等 。HTTP 自动跳转 HTTPS，无必要配置 H2 协议。NaiveProxy 的 HTTPS 代理最好仅使用非 AES 算法的密码套件。
1、不必关心 Caddy 反代‘流媒体’日志报错问题，这个是 [BUG](https://github.com/XTLS/Xray-core/issues/3856)。你配置 1 的所有应用无必要参考 [Xray(M+F+B+D+G+A)+Caddy(N)](https://github.com/lxhao61/integrated-https://githuexamples/tree/main/Xray(M%2BF%2BB%2BD%2BG%2BA)%2BCaddy(N))， 推荐参考  [Xray(M+B+D+G+A)+Caddy(N+T)](https://github.com/lxhao61/integrated-examples/tree/main/Xray(M%2BB%2BD%2BG%2BA)%2BCaddy(N%2BT)) 配置。
2、你配置 2 的 Xray 中 "maxVersion": "1.3" 修改为"minVersion": "1.3"，其上增加 "alpn": ["h3"]。另外配置 2 的 NaiveProxy 与 SplitHTTP 的 HTTP/3 代理仅二选一，即 UDP 443 定向配置转发给其中一个。
3、NaiveProxy 或 SplitHTTP 的 HTTP/3 代理无法使用，推测防火墙没有开放 UDP 443 端口及没有增加 UDP 接收缓冲区大小。

**Comment by rannj** [comment]
感谢回复！我没有使用CDN的需求，根据你的建议，我准备弃用SplitHTTP的相关配置。

**Reply by lxhao61** [reply]
SplitHTTP 的主要优势是可解决 CDN 不支持 WebSocket/HTTPUpgrade、gRPC 问题。
